site/app/libraries/database/DatabaseQueries.php

```diff
@@ -4919,24 +4920,29 @@ public function getPeerGradingAssignmentsForGrader($grader_id) {
      * Instructors may access all of their courses
      * Inactive courses may only be accessed by the instructor
      *
-     * @param  string $user_id
-     * @param  bool   $archived
+     * If $dropped is true, we return null section courses as well.
+     *
+     * @param  string $user_id  User Id of user we're getting courses for.
+     * @param  bool   $archived True if we want archived courses.
+     * @param  bool   $dropped  True if we want null section courses.
      * @return Course[] archived courses (and their details) accessible by $user_id
      */
-    public function getCourseForUserId($user_id, bool $archived = false): array {
+    public function getCourseForUserId($user_id, bool $archived = false, bool $dropped = false): array {
+        $include_archived = "AND c.status=1";
         if ($archived) {
-            $extra = "AND c.status=2 AND u.user_group=1";
+            $include_archived = "AND c.status=2 AND u.user_group=1";
         }
-        else {
-            $extra = "AND c.status=1";
+        $force_nonnull = "";
+        if ($dropped) {
+            $force_nonnull = "NOT";
         }
 
         $query = <<<SQL
 SELECT t.name AS term_name, u.term, u.course, u.user_group, u.registration_section
 FROM courses_users u
 INNER JOIN courses c ON u.course=c.course AND u.term=c.term
 INNER JOIN terms t ON u.term=t.term_id
-WHERE u.user_id=? ${extra} AND (u.registration_section IS NOT NULL OR u.user_group<>4)
+WHERE u.user_id=? ${include_archived} AND ${force_nonnull} (u.registration_section IS NOT NULL OR u.user_group<>4)
```

Comment:
There are two places where registration section is changed; in the "submitty" database and the specific "submitty_semester_course" database. This PR (as well as all previous PRs) only cause users to see their dropped course as being "dropped" (eg. not in "My Courses") if the former is changed; if the latter is changed, the user is still denied access but does not see it until they attempt to access the course.

Reply:
Highlighting this line because this is where the dependence on the overall submitty database comes in (the registration status on the course database is not considered here).